### PR TITLE
Adapt split of tf and pytorch to Numpy's

### PR DESCRIPTION
**Pull request by pchauchat**
This PR adresses #567 : `split` functions are defined in Pytorch and Tensorflow to match the behavior of Numpy's one. This is done by inferring the right arguments to pass to them from the arguments that are normallygiven when using Numpy.

### CODE REVIEW

File: geomstats/_backend/pytorch/__init__.py [hunk 1]
```diff
@@ -92,8 +92,17 @@ def empty(shape, dtype=float64):
     return torch.empty(*shape, dtype=dtype)
 
 
-def split(ary, indices_or_sections, axis=0):
-    return torch.split(ary, indices_or_sections, dim=axis)
+def split(x, indices_or_sections, axis=0):
+    if isinstance(indices_or_sections, int):
+        indices_or_sections = indices_or_sections // x.shape[axis]
```
Comment (nguigs) [inline geomstats/_backend/pytorch/__init__.py:97]: shouldn't it be the inverse? `torch.split(x, k)` will return n // k (+1) arrays of size k (except the last one) while `numpy.split(x, k)` would return k arrays of size n // k, where `n = x.shape[axis]`, and throw an error if `n % k != 0`

File: tests/test_product_manifold.py [hunk 2]
```diff
@@ -28,15 +28,13 @@ def test_dimension(self):
         result = self.space_vector.dim
         self.assertAllClose(result, expected)
 
-    @geomstats.tests.np_and_pytorch_only
```
Comment (nguigs) [inline tests/test_product_manifold.py:31]: :tada:

File: geomstats/_backend/pytorch/__init__.py (hunk 1 above)
Comment (pchauchat) [reply geomstats/_backend/pytorch/__init__.py:97]: True, I had mixed them up, and everything was working fine since we were in a case were `indices_or_sections // x.shape[axis] = 1` ...

File: geomstats/_backend/pytorch/__init__.py [hunk 3]
```diff
@@ -604,6 +613,8 @@ def copy(x):
 
 
 def cumsum(x, axis=None):
+    if not torch.is_tensor(x):
+        x = array(x)
```
Comment (ninamiolane) [inline geomstats/_backend/pytorch/__init__.py:617]: Yes, I had to add a similar condition in less_equal for torch. Why using `x = array(x)` here and not `x = torch.tensor(x)`, which could be faster? Do we need the logic of the `array` function?

File: geomstats/geometry/hypersphere.py [hunk 4]
```diff
@@ -301,6 +301,7 @@ def random_uniform(self, n_samples=1):
         while True:
             norms = gs.linalg.norm(samples, axis=1)
             indcs = gs.isclose(norms, 0.0)
+            indcs = gs.cast(indcs, gs.int32)
```
Comment (ninamiolane) [inline geomstats/geometry/hypersphere.py:304]: Since `np.sum` works with booleans, could we modify `gs.sum`, in order to do the cast to `gs.int32` in the tf/torch's backend function and not in the main code?

File: tests/test_product_manifold.py [hunk 5]
```diff
@@ -28,15 +28,13 @@ def test_dimension(self):
...
     def test_random_and_belongs_matrix(self):
         n_samples = 5
         data = self.space_matrix.random_uniform(n_samples)
         result = self.space_matrix.belongs(data)
         expected = gs.array([True] * n_samples)
         self.assertAllClose(result, expected)
 
-    @geomstats.tests.np_only
```
Comment (ninamiolane) [inline tests/test_product_manifold.py:39]: Wooohoo! 🎉

File: geomstats/_backend/pytorch/__init__.py (hunk 3 above)
Comment (pchauchat) [reply geomstats/_backend/pytorch/__init__.py:617]: It depends on what kind of input this should accept. Say you want to do something like `np.cumsum([np.ones(3), np.ones(3)])`, then you will need the `array` function. Otherwise, `x = torch.tensor(x)` would be fine.

Comment (ninamiolane) [reply geomstats/_backend/pytorch/__init__.py:617]: Got it, thanks for explaining.
